packages/tools/devtools/devtools-view/src/DevtoolsView.tsx

```diff
@@ -183,10 +186,40 @@ export function DevtoolsView(): React.ReactElement {
 		};
 	}, [messageRelay, setSupportedFeatures]);
 
+	// Manage the query timeout
+	React.useEffect(() => {
+		if (supportedFeatures === undefined) {
+			// If we have queried for the supported feature list but have not received
+			// a response yet, queue a timer.
+			const timeout = setTimeout(() => {
```

Comment:
I'm not super familiar with the timeout APIs - how does this timeout object relate to `queryTimer`? Based on my limited understanding, it looks like we have two separate/unrelated timer objects being used.